README.md

```diff
@@ -86,6 +86,30 @@ rl.on('close', () => {
 });
 ```
 
+**C#**
+```C#
+private static readonly Lazy<HashSet<string>> _emailBlackList =
+  new Lazy<HashSet<string>>(
+    () =>
+    {
+      var result = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+      using (var reader = new StreamReader("disposable_email_blacklist.conf"))
+      {
+        string line;
+        while ((line = reader.ReadLine()) != null)
+          if (!string.IsNullOrWhiteSpace(line) && !line.TrimStart().StartsWith("//"))
+            result.Add(line);
+      }
+      return result;
+    });
+
+...
+
+var addr = new MailAddress(email);
+if (_emailBlackList.Value.Contains(addr.Host)))
```

Comment:
You should compare with top level domain (2 or 3 levels), not the entire domain.

For example, if the disposable email is `test.spamtrap.com`, there will be only `spamtrap.com` in the black list, as we black list the top domain, for avoiding every subdomain to be declared in the file (that could be a huge task an hard to maintain, as it's very easy to create a subdomain).

There are some case where the domain can have three levels (for example, `spamtrap.co.uk`)...


The others langage usage samples are not all updated as the switch to top level domain blacklist is recent.

Reply:
See discussion on #45 about the domain levels.